app/CMakeLists.txt

```diff
@@ -52,119 +39,94 @@ option(SANITIZE_THREAD    "Sanitize Thread"        OFF)
-option(LIBIGL_USE_STATIC_LIBRARY          "Use libigl as static library" OFF)
-option(LIBIGL_WITH_COMISO            "Use CoMiso"                   OFF)
-option(LIBIGL_WITH_EMBREE            "Use Embree"                   OFF)
-option(LIBIGL_WITH_OPENGL            "Use OpenGL"                   OFF)
-option(LIBIGL_WITH_OPENGL_GLFW       "Use GLFW"                     OFF)
-option(LIBIGL_WITH_OPENGL_GLFW_IMGUI "Use ImGui"                    OFF)
-option(LIBIGL_WITH_PNG               "Use PNG"                      OFF)
-option(LIBIGL_WITH_TETGEN            "Use Tetgen"                   OFF)
-option(LIBIGL_WITH_TRIANGLE          "Use Triangle"                 OFF)
-option(LIBIGL_WITH_PREDICATES        "Use exact predicates"         ON)
-option(LIBIGL_WITH_XML               "Use XML"                      OFF)
+option(LIBIGL_USE_STATIC_LIBRARY        "Use libigl as static library" OFF)
+option(LIBIGL_WITH_COMISO            	"Use CoMiso"                   OFF)
+option(LIBIGL_WITH_EMBREE            	"Use Embree"                   OFF)
+option(LIBIGL_WITH_OPENGL            	"Use OpenGL"                   OFF)
+option(LIBIGL_WITH_OPENGL_GLFW       	"Use GLFW"                     OFF)
+option(LIBIGL_WITH_OPENGL_GLFW_IMGUI 	"Use ImGui"                    OFF)
+option(LIBIGL_WITH_PNG               	"Use PNG"                      OFF)
+option(LIBIGL_WITH_TETGEN            	"Use Tetgen"                   OFF)
+option(LIBIGL_WITH_TRIANGLE          	"Use Triangle"                 OFF)
+option(LIBIGL_WITH_PREDICATES        	"Use exact predicates"         ON)
+option(LIBIGL_WITH_XML               	"Use XML"                      OFF)
 
 # Sanitizers
-if(FLOAT_TETWILD_WITH_SANITIZERS)
-	list(APPEND CMAKE_MODULE_PATH ${FLOAT_TETWILD_EXTERNAL}/sanitizers-cmake/cmake)
+if(TWTK_APPS_WITH_SANITIZERS)
+	list(APPEND CMAKE_MODULE_PATH ${TWTK_APPS_EXTERNAL}/sanitizers-cmake/cmake)
 endif()
 
-# Setup dependencies
-# include(FloatTetwildDependencies)
 
 ################################################################################
-# FloatTetwild library
+# FloatTetwild
 ################################################################################
 
-#find_package(GMPfTetWild)
-#IF(NOT ${GMP_FOUND})
-#	MESSAGE(FATAL_ERROR "Cannot find GMP")
-#ENDIF()
-
-# find_package(MPFR)
-# IF(NOT ${MPFR_FOUND})
-# 	MESSAGE(FATAL_ERROR "Cannot find MPFR")
-# ENDIF()
-
-# add_library() can only be called without any source since CMake 3.11 ...
-#set(SOURCES
-#		TetWild.cpp TetWild.h
-#				common.h
-#				Parameters.h
-#				Envelope.h
-#		)
-#
-#prepend_current_path(SOURCES)
-#wildmeshing_toolkit_copy_headers(${SOURCES})
-#wildmeshing_toolkit_set_source_group(${SOURCES})
-#target_sources(${PROJECT_NAME} PRIVATE ${SOURCES})
-
-add_library(${PROJECT_NAME} #SOURCES)
+add_library(TetWild
+		tetwild/main.cpp
+
 		external/Exception.h
 		external/MshSaver.cpp
 		external/MshSaver.h
 
-		TetWild.cpp
-		TetWild.h
+		tetwild/TetWild.cpp
+		tetwild/TetWild.h
 
 		Logger.cpp
```

Comment:
Consider making a `app/shared/` library for shared utils.

Reply:
yeah, but also root folder is ok, if you have strong feelings for folder, I will add

Reply:
`app/shared/` should be a proper CMake target, rather than just compiling the sources all over again. Having it as a subfolder prevents accidental usage of non-dependencies (e.g. `Logger.cpp` including something from `tetwild/*.h`).

Reply:
(By having a proper cmake target you avoid repeating yourself when adding common files between apps).